### PR TITLE
Blueflood HTTP Timeouts

### PR DESCRIPTION
**Pull request by Nathan-Fenner**
This PR changes the way that Blueflood handles timeouts. In particular:

* Blueflood's `Config` no longer has a `Timeout` field. Instead, you should set the timeout for a *query* (not just fetches to Blueflood) through the `Timeout` field in the `ExecutionContext`.
* Blueflood now respects the `Timeout` object passed by to it through `EvaluationContext` and cancels requests when it ends.
* Blueflood was organized slightly (mostly reordering functions).
* Testing support's `HTTPClient` now supports a `Do` function that just calls `Get` for mocking purposes.
* The `ParallelQueue` in `tasks` has been updated to use a passed-in `*tasks.Timeout` instead of creating its own from a `time.Duration`.

@drcapulet

### CODE REVIEW

File: timeseries/blueflood/blueflood.go [hunk 1]
```diff
@@ -279,188 +267,48 @@ func (b *Blueflood) constructURL(metric api.TaggedMetric, interval api.Interval,
 	return result, nil
 }
 
-// performFetch is a synchronous method that fetches the given URL.
-func (b *Blueflood) performFetch(queryURL *url.URL) (queryResponse, error) {
-	resp, err := b.config.HTTPClient.Get(queryURL.String())
+type httpClient interface {
+	// our own client to mock out the standard golang HTTP Client.
+	Get(string) (resp *http.Response, err error)
```
Comment (drcapulet) [inline timeseries/blueflood/blueflood.go]: No need to name the return parameters here.

**Comment by drcapulet** [comment]
Overall LGTM with one comment, any thoughts on using the `golang.org/x/net/context` instead of our timeout solution?

**Comment by Nathan-Fenner** [comment]
I think using the `net/context` instead of `Timeout` would be a good idea.

**Comment by drcapulet** [comment]
LGTM, let's switch to `net/context` in a follow up PR.

**Pull request merged**